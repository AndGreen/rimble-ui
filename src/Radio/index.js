import React from 'react'
import styled from 'styled-components'

import theme from '../theme'

import RadioButtonChecked from 'rmdi/lib/RadioButtonChecked'
import RadioButtonUnchecked from 'rmdi/lib/RadioButtonUnchecked'

const StyledWrapper = styled.div`
  & {
    position: relative;
    display: inline-block;
    vertical-align: middle;
  }

  > input {
    position: relative;
    display: block;
    appearance: none;
    background: white;
    height: 1.5rem;
    width: 1.5rem;
    margin: 0;
    font-weight: bold;
  }

  > svg {
    position: absolute;
    z-index: 0;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    height: 100%;
    width: 100%;
    pointer-events: none;
    fill: ${theme.colors.grey};
  }

  > svg[name="checked"] {
    display: none;
    z-index: 1;
    fill: ${theme.colors.primary};
  }

  > input:checked ~ svg[name="checked"] {
    display: block;
  }
`

const Radio = (props) => (
  <StyledWrapper className={props.className}>
    <input type="radio" {...props} />
    <RadioButtonChecked name='checked' />
    <RadioButtonUnchecked />
  </StyledWrapper>
)

Radio.defaultProps = {
  theme: theme
}

Radio.displayName = 'Radio'

export default Radio
