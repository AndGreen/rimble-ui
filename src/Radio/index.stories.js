import React from 'react'

import { storiesOf } from '@storybook/react'

import Radio from '../Radio'

storiesOf('Form/Radio', module).add('default', () => (
  <div>
    <label htmlFor=""><Radio id="blue" name="pill"/>Blue Pill</label>
    <label htmlFor=""><Radio id="red" name="pill" />Red Pill</label>
  </div>
))
