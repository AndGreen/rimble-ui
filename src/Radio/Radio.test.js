import Radio from './'

describe('Radio component sanity', () => {
  it('has name', () => {
    expect(Radio.displayName).toBe('Radio')
  })

  it('matches default snapshot', () => {
    const component = render(<Radio />)
    expect(component).toMatchSnapshot()
  })
})
