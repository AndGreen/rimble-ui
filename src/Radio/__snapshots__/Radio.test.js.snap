// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component sanity matches default snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: 1;
}

.c0 {
  box-sizing: border-box;
}

.c2 {
  margin-left: 4px;
  margin-right: 8px;
  color: copyColor;
  font-size: 14px;
  font-size: 14px;
  font-family: sansSerif;
  font-weight: 3;
  text-align: left;
  line-height: copy;
}

.c2 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: inline-block;
}

.c1 > input {
  position: relative;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 1.5rem;
  width: 1.5rem;
  margin: 0;
}

.c1 > svg {
  position: absolute;
  z-index: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  pointer-events: none;
  fill: #CCC;
}

.c1 > svg[name="checked"] {
  display: none;
  z-index: 1;
  fill: #4E3FCE;
}

.c1 > input:checked ~ svg[name="checked"] {
  display: block;
}

<label
  class="c0"
  display="flex"
  opacity="1"
>
  <div
    class="c1"
  >
    <input
      theme="[object Object]"
      type="checkbox"
    >
      
    </input>
    <svg
      class=""
      fill="currentcolor"
      height="24"
      name="checked"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      >
        
      </path>
    </svg>
    <svg
      class=""
      fill="currentcolor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      >
        
      </path>
    </svg>
  </div>
  <div
    class="c2"
    color="copyColor"
    font-family="sansSerif"
    font-size="1"
    font-weight="3"
  >
    
  </div>
</label>
`;
